pkgs/by-name/ma/maltego/package.nix

```diff
@@ -20,16 +20,16 @@ stdenv.mkDerivation (finalAttrs: {
   };
 
   postPatch = ''
-      substituteInPlace bin/maltego \
-            --replace /usr/bin/awk ${lib.getExe gawk}
+    substituteInPlace bin/maltego \
+      --replace /usr/bin/awk ${lib.getExe gawk}
   '';
 
-    desktopItems = [
+  desktopItems = [
     (makeDesktopItem {
-      name = finalAttrs.pname;
+      name = "maltego";
       desktopName = "Maltego";
-      exec = finalAttrs.meta.mainProgram;
-      icon = finalAttrs.pname;
+      exec = "maltego";
```

Comment:
@SuperSandro2000 @Janik-Haag I believe you still had a bit of discussion left about this line in particular from the previous thread?

Reply:
Mmm why removing `icon` attribute?

Reply:
No, rather about whether `exec` should be converted to use a string literal.

In contrast to the icon which could be named arbitrarily and does not depend on the package name, the `exec` should always be the same as `mainProgram` for as long as the package only has a single executable. It might make sense to keep this depending on `finalAttrs.meta.mainProgram`, but I don't really have any strong opinions about this.

On another note, it would be nice if we could pinpoint this stuff in the contributing docs, but I don't think that's a topic for this particular thread.

Reply:
```nix
      exec = finalAttrs.meta.mainProgram;
```
I feel having is preferable and if no one else says hard no with a good reason I would like to keep it as is.

Reply:
> On another note, it would be nice if we could pinpoint this stuff in the contributing docs, but I don't think that's a topic for this particular thread.

Probably, feel free to do a pr/discourse post. I'm currently too swamped for stuff like that because of $life